### PR TITLE
Remove CrossPlatform.sln

### PR DESCRIPTION
**Pull request by agocke**
The core work here is that we were pulling reference assemblies out of a folder with the name `RoslynTools.ReferenceAssemblies`. A while ago NuGet switched to unpacking all folders with lowercase paths. On Windows this continues to work fine as the filesystem APIs are case-insensitive, but this fails on Linux. By matching casing we can build both net461 and netcoreapp targets on any operating system. The inability to do this was the primary reason for CrossPlatform.sln, so there's now no reason for the solution and we can switch to using Compilers.sln, which is a more reasonable superset of everything in CrossPlatform.sln.

### CODE REVIEW

File: build/Targets/Settings.props [hunk 1]
```diff
@@ -48,7 +48,7 @@
     <ToolsetCompilerPackageVersion Condition="'$(OS)' == 'Windows_NT'">$(MicrosoftNetCompilersVersion)</ToolsetCompilerPackageVersion>
     <ToolsetCompilerPackageVersion Condition="'$(OS)' != 'Windows_NT'">$(MicrosoftNETCoreCompilersVersion)</ToolsetCompilerPackageVersion>
     <ToolsetCompilerPropsFilePath>$(NuGetPackageRoot)/$(ToolsetCompilerPackageName.ToLower())/$(ToolsetCompilerPackageVersion)/build/$(ToolsetCompilerPackageName).props</ToolsetCompilerPropsFilePath>
-    <TargetFrameworkRootPath>$(NuGetPackageRoot)\RoslynTools.ReferenceAssemblies\$(RoslynToolsReferenceAssembliesVersion)\tools\Framework</TargetFrameworkRootPath>
+    <TargetFrameworkRootPath>$(NuGetPackageRoot)\roslyntools.referenceassemblies\0.1.1\tools\framework</TargetFrameworkRootPath>
```
Comment (jcouv) [inline build/Targets/Settings.props]: Is this version number hardcoded on purpose?

File: CrossPlatform.sln [hunk 2]
```diff
@@ -1,358 +0,0 @@
-
```
Comment (jcouv) [inline CrossPlatform.sln]: @agocke If we remove CrossPlatform.sln, I assume this means `dotnet build Roslyn.sln` and `dotnet build Compilers.sln` work on Linux?

File: build/Targets/Settings.props (hunk 1 above)
Comment (agocke) [reply build/Targets/Settings.props]: No, I must have missed that when I was prototyping. I'll replace with the property.

**Comment by tmat** [comment]
👍

**Comment by jcouv** [comment]
Could you update the "build for linux" [docs](https://github.com/dotnet/roslyn/blob/master/docs/infrastructure/cross-platform.md) too?

File: build/Targets/Settings.props [hunk 3]
```diff
@@ -61,7 +61,7 @@
     <ToolsetCompilerPackageVersion Condition="'$(OS)' == 'Windows_NT'">$(MicrosoftNetCompilersVersion)</ToolsetCompilerPackageVersion>
     <ToolsetCompilerPackageVersion Condition="'$(OS)' != 'Windows_NT'">$(MicrosoftNETCoreCompilersVersion)</ToolsetCompilerPackageVersion>
     <ToolsetCompilerPropsFilePath>$(NuGetPackageRoot)/$(ToolsetCompilerPackageName.ToLower())/$(ToolsetCompilerPackageVersion)/build/$(ToolsetCompilerPackageName).props</ToolsetCompilerPropsFilePath>
-    <TargetFrameworkRootPath>$(NuGetPackageRoot)\RoslynTools.ReferenceAssemblies\$(RoslynToolsReferenceAssembliesVersion)\tools\Framework</TargetFrameworkRootPath>
+    <TargetFrameworkRootPath>$(NuGetPackageRoot)\roslyntools.referenceassemblies\$(RoslynToolsReferenceAssembliesVersion)\tools\framework</TargetFrameworkRootPath>
```
Comment (jaredpar) [inline build/Targets/Settings.props]: How did I miss this????????

File: src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs [hunk 4]
```diff
@@ -198,7 +198,7 @@ private static (T Result, string Output) UseTextWriter<T>(Encoding encoding, Fun
             var client = ServerUtil.CreateBuildClient(language);
             client.TimeoutOverride = Timeout.Infinite;
 
-#if NET46
+#if NET461
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs]: Why did this change?

File: src/Compilers/Server/VBCSCompilerTests/VBCSCompilerTests.csproj [hunk 5]
```diff
@@ -10,7 +10,7 @@
     <RootNamespace>Microsoft.CodeAnalysis.CompilerServer.UnitTests</RootNamespace>
     <AssemblyName>Roslyn.Compilers.CompilerServer.UnitTests</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <TargetFrameworks>$(RoslynPortableTargetFrameworks46)</TargetFrameworks>
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompilerTests/VBCSCompilerTests.csproj]: Why did we change this?

**Pull request merged**